Update UaaS Documentation (#960)
centrapay/centrapay-docs: src/content/guides/integrating-third-party-asset.mdx

An idempotency key is used for `HTTP POST` requests. Centrapay **does not** guarantee endpoints will be called only once per transaction. It is expected that you will enforce transaction idempotency. In the event that the idempotency is violated, Centrapay expects a `200 OK` response as described in the endpoint specification.

### Errors

> We expect to receive a 2XX response from all endpoints for both successful and failed transactions unless an unexpected error has occurred in your system. Centrapay will retry a transaction attempt when an HTTP response status code ≥ 500 is thrown.

Reviewer: ```suggestion
A 2XX response should be returned from all endpoints for both successful and failed transactions unless an unexpected error has occurred in your system. Centrapay will retry a transaction attempt when an HTTP response status code ≥ 500 is thrown.
```

> We expect to receive a 2XX response from all endpoints for both successful and failed transactions unless an unexpected error has occurred in your system. Centrapay will retry a transaction attempt when an HTTP response status code ≥ 500 is thrown.

Reviewer 2: ```suggestion
We expect to receive a 2XX response from all endpoints for both successful and failed transaction attempts unless an unexpected error has occurred in your system. Centrapay will retry a transaction attempt when an HTTP response status code ≥ 500 is thrown.
```

> Centrapay will also use the [Get Transaction Endpoint](#get-transaction-endpoint) to resolve transactions if errors are thrown.

Reviewer: ```suggestion
If a 4XX error is thrown and the payment is not resolved Centrapay will use the [Get Transaction Endpoint](#get-transaction-endpoint) to resolve transaction.
```

> Centrapay will also use the [Get Transaction Endpoint](#get-transaction-endpoint) to resolve transactions if errors are thrown.

Reviewer 2: @AndyClifford Suggestion: Similar to my other comment - avoid tacking things on with `also`

Change: I think this section should describe how to use the `failed` status + `failureReason` too

After initiating a payment transaction, the status may be `pending`. This endpoint will be used to poll the status of the transaction attempt. Polling will continue until either the transaction attempt status is `successful` or `failed`, or the Centrapay Payment Request is no longer payable (e.g. it has expired).

> We will also use this endpoint to resolve transactions if errors are being thrown from your pay endpoint. If the transaction does not exist in your system we expect a 2XX response with an empty object as the body.

Reviewer: ```suggestion
We will also use this endpoint to resolve transactions if we are unsure of the status of a payment attempt. If the transaction does not exist in your system a 2XX response with an empty object as the body should be returned.
```

> We will also use this endpoint to resolve transactions if we are unsure of the status of a transaction attempt. If the transaction does not exist in your system a 2XX response with an empty object as the body should be returned.

Reviewer 2: @AndyClifford These changes feel like they've been tacked on as an after thought (multiple uses of `also`). 

Suggestion: Make important details clearer and make this endpoint description more succinct. I'd consider at rewriting this entire endpoint description.

> This endpoint is used to poll the Transaction Attempt status when it is `pending` or unknown due to an unexpected behavior. 
>
> Polling will continue until either the transaction attempt status is `successful` or `failed`, or the Centrapay Payment Request is no longer payable (e.g. it has expired).
>
> You should return an empty 2XX response if the `transactionId` does not exist in your system.
